memcache/memcache.go

```diff
@@ -711,3 +729,244 @@ func (c *Client) incrDecr(verb, key string, delta uint64) (uint64, error) {
 	})
 	return val, err
 }
+
+type MetaSetItem struct {
+	Key    string
+	Flags  MetaSetFlags
+	Length int32
+	Value  []byte
+}
+
+type MetaResponseMetadata struct {
+	Value                 []byte
+	CasId                 *uint64
+	Key                   *string
+	OpaqueValue           *string
+	TTLRemainingInSeconds *int32
+}
+
+type MetaSetFlags struct {
+	// use if the key provided is base 64 encoded
+	// equivalent to the b flag
+	IsKeyBase64 bool
+
+	// use if you wish to see the cas token as a part of the response
+	// equivalent to the c flag
+	ReturnCasTokenInResponse bool
+
+	// use if you provide a cas token with CompareCasTokenToUpdateValue attribute and it is older than the item's CAS
+	// note: only functional when combined with the CompareCasTokenToUpdateValue attribute
+	// equivalent to the I flag
+	Invalidate bool
+
+	// use if you want to get the key as a part of the response
+	// equivalent to the k flag
+	ReturnKeyInResponse bool
+
+	// equivalent to the q flag
+	UseNoReplySemanticsForResponse bool
+
+	// use if only want to store a value if the supplied token matches the current CAS token of the item
+	// equivalent to the C<token> flag
+	CompareCasTokenToUpdateValue *uint64
+
+	// use if you want to set client flags to a token
+	// equivalent to the F<token> flag
+	ClientFlagToken *uint32
+
+	// use if you want to consume a token and copy back with a response
+	// equivalent to the O<token> flag
+	OpaqueToken *string
+
+	// use if you want to set the TTL for the item
+	// equivalent to the T<token> flag
+	UpdateTTLToken *uint32
+
+	// use if you want to switch modes
+	// E: "add" command. LRU bump and return NS if item exists, else add
+	// A: "append" command. If item exists, append the new value to its data
+	// P: "prepend" command. If item exists, prepend the new value to its data
+	// R: "replace" command. Set only if item exists, replace its value
+	// S: "set" command. The default mode, added for completeness
+	// equivalent to the M<token> flag
+	SetModeToken *string
```

Comment:
Since this is a free string we need some sort of validation here if a user enters some invalid string. Check if Go supports some sort of enums so that we can enforce this.

Reply:
I created an enum for this type now. Let me know what you think of the approach!